### PR TITLE
feat: extensions support and add path parameters to operation params

### PR DESCRIPTION
**Pull request by khizunov**
- Extensions support in all supported objects
- Add global path parameters to operation

### CODE REVIEW

**Comment by khizunov** [comment]
@manchenkoff

File: src/openapi_parser/builders/request.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import logging
 
 from . import ContentBuilder
-from .common import extract_typed_props, PropertyMeta
+from .common import extract_typed_props, PropertyMeta, extract_extension_attributes
```
Comment (manchenkoff) [inline src/openapi_parser/builders/request.py:4]: looks like this import is not used anywhere

**Review by manchenkoff — CHANGES_REQUESTED** [review]
Hey @khizunov, thanks a lot for the contribution! 🔥 
Left one code-related minor comment, but there are some build issues.
Could you also add some unit tests as well? At least to check more complicated logic like passing extensions to each oauth flow and path parameters inheritance.

**Comment by khizunov** [comment]
> Hey @khizunov, thanks a lot for the contribution! 🔥 Left one code-related minor comment, but there are some build issues. Could you also add some unit tests as well? At least to check more complicated logic like passing extensions to each oauth flow and path parameters inheritance.

Hey @manchenkoff sure not problem at all, will do 👍

**Comment by manchenkoff** [comment]
Closes #42

**Comment by khizunov** [comment]
@manchenkoff good catch on Oauth flows 💯 
Fixed the build, working on tests

**Comment by khizunov** [comment]
@manchenkoff added test cases for extensions
for the path parameters inheritance the cases are already there in path builder test and runner test.
LMK if you want me to add some other cases

File: tests/test_runner.py [hunk 2]
```diff
@@ -13,4 +13,4 @@ def swagger_specification():
 def test_run_parser(swagger_specification: Specification):
     actual_specification = parse('tests/data/swagger.yml')
 
-    assert swagger_specification == actual_specification
```
Comment (manchenkoff) [inline tests/test_runner.py:16]: just out of curiosity, why did you change the order of comparison? 🙂

**Review by manchenkoff — APPROVED** [review]

File: tests/test_runner.py (hunk 2 above)
Comment (khizunov) [reply tests/test_runner.py:16]: @manchenkoff pytest assert message uses left as `actual` :-)

File: src/openapi_parser/builders/path.py [hunk 3]
```diff
@@ -42,4 +42,13 @@ def _build_path(self, url: str, data: dict) -> Path:
             if method.value in data
         ]
 
+        if attrs.get("parameters"):
```
Comment (manchenkoff) [inline src/openapi_parser/builders/path.py:45]: > @manchenkoff added test cases for extensions for the path parameters inheritance the cases are already there in path builder test and runner test. LMK if you want me to add some other cases

I've checked [path builder test](https://github.com/manchenkoff/openapi3-parser/blob/c0c91ec77b94cb0820f068789f6bd715e8ae07fd/tests/builders/test_path_builder.py#L36), but it looks like Operation  doesn't have parameters from Path, is it expected behavior for this your change?

File: tests/test_runner.py (hunk 2 above)
Comment (manchenkoff) [reply tests/test_runner.py:16]: oh, I see, thx :D

File: src/openapi_parser/builders/path.py (hunk 3 above)
Comment (khizunov) [reply src/openapi_parser/builders/path.py:45]: I noticed this as well, and it appears that the same operation object is present in both the expected and actual results. I debugged it and saw same parameter in operation as well (3 case). Let me try to understand it better.

Comment (khizunov) [reply src/openapi_parser/builders/path.py:45]: @manchenkoff 
In fact, the test is somewhat challenging :-) The same operation_object (reference) is utilized for the mock_builder and as the expected object. Therefore, when the path builder constructs the actual object, it updates the parameters for both. 
I can fix the test. (the behavior is expected)

Comment (manchenkoff) [reply src/openapi_parser/builders/path.py:45]: yikes 😄 yes, please update the test once you have time

Comment (khizunov) [reply src/openapi_parser/builders/path.py:45]: done

**Review by manchenkoff — APPROVED** [review]

**Comment by manchenkoff** [comment]
I will publish new version later today, thanks @khizunov!

**Comment by manchenkoff** [comment]
published as v1.1.17
